src/Yaapii.Atoms/Map/Sorted.cs

```diff
@@ -0,0 +1,178 @@
+            : this(pairs, new ComparerFunc<KeyValuePair<Key, Value>>(compare))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given key comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<Key, Value>> pairs, IComparer<Key> cmp)
+            : this(pairs, new KeyComparer<Key, Value>(cmp))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing elements</param>
+        public Sorted(IEnumerable<KeyValuePair<Key, Value>> pairs, IComparer<KeyValuePair<Key, Value>> cmp)
+            : base(
+                () =>
+                {
+                    var items = new List<KeyValuePair<Key, Value>>(pairs);
+                    items.Sort(cmp);
+                    return new Map.Of<Key, Value>(items);
+                },
+                false
+            )
+        { }
+    }
+
+    public sealed class Sorted<Value> : Map.Envelope<Value>
+    {
+        /// <summary>
+        /// Sorts the given map with the default comperator of the key
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs)
+            : this(pairs, Comparer<string>.Default)
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given compare function
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="compare">Function to compare two elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, Func<KeyValuePair<string, Value>, KeyValuePair<string, Value>, int> compare)
+            : this(pairs, new ComparerFunc<KeyValuePair<string, Value>>(compare))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given key comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing keys</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<string> cmp)
+            : this(pairs, new KeyComparer<string, Value>(cmp))
+        { }
+
+        /// <summary>
+        /// Sorts the given map with the given comparer
+        /// </summary>
+        /// <param name="pairs">Map elements to be sorted</param>
+        /// <param name="cmp">Comparer comparing elements</param>
+        public Sorted(IEnumerable<KeyValuePair<string, Value>> pairs, IComparer<KeyValuePair<string, Value>> cmp)
+            : base(
+                () =>
+                {
+                    var items = new List<KeyValuePair<string, Value>>(pairs);
```

Comment:
Can't we use a SortedDictionary? Background is that I would like us to not copy elements if not needed. I hope SortedDictionary has a ctor accepting another Dict.

Reply:
`SortedDictionary` has a constructor accepting a Dictionary but only a `Comparer<Key>` without value. #346 specifically requests a sorted dictionary by value.

`List<>` is not immutable and only keeps a list of references. Elements are not copied. (Even `List<>.AsReadOnly()` forbids only Add, Remove, ... but still can contain mutable objects)

Reply:
List will enumerate the items, that is the main issue I want to point out. I wanted to avoid this. The main UseCase for the library will be sorting by key, this should be possible without building all the values. The copy ctor of list will instead need build all value as it will enumerate. This will in most cases be rejected by LazyDict. 
It is not necessary for sorting by key. Can you please find a solution which avoids building all values if it is not needed?

Reply:
Regarding Immutability, ListOf and c# ReadonlyList are both immutable in our understanding. Immutability means for us that the coordinates of an object do not change - in other words, no references inside the object are ever replaced. 

However, being Readonly and being Immutable are two things, we can build a non-readonly object which is immutable. It would be immutable only because its private members are all readonly.

Reply:
`SortedDictionary` uses `.GetEnumerator()` which is "building all values".